File: playwright/nodejs/tests/otpEmail.spec.js
require('dotenv').config();

const MailosaurClient = require('mailosaur');
const { expect, test } = require('@playwright/test');

// Instantiate Mailosaur client with api key
const mailosaur = new MailosaurClient(process.env.MAILOSAUR_API_KEY);
const serverId = process.env.MAILOSAUR_SERVER_ID;

test.describe('One time passcode - Email', () => {
  test('Retrieve one time passcode', async ({ page }) => {
    // Random test email address (this uses a catch-all pattern)
    const randomString = (Math.random() + 1).toString(36).substring(7);
    const emailAddress = `${randomString}@${serverId}.mailosaur.net`;

    // 1 - Request a one time passcode
    await page.goto(`https://example.mailosaur.com/otp`);
    await page.fill('#email', emailAddress);
    await page.click('button[type="submit"]');

    // 2 - Create the search criteria for the email
    // https://mailosaur.com/docs/api/messages#4-search-for-messages
    const searchCriteria = {
      sentTo: emailAddress,
    };

    // 3 - Get the email from Mailosaur using the search criteria
    const email = await mailosaur.messages.get(serverId, searchCriteria);

    expect(email.subject).toEqual('Here is your access code for ACME Product');

    // 4 - Extract the passcode from the email
    // https://mailosaur.com/docs/test-cases/codes
    const passcode = email.html.codes[0];

    console.log(`\nEmail otp code - ${passcode.value}`); // "564214"
  });
});

File: selenium/ruby/spec/otp_email_spec.rb
# frozen_string_literal: true

require 'mailosaur'
require 'selenium-webdriver'

RSpec.describe 'Otp email' do
  before do
    api_key = ENV['MAILOSAUR_API_KEY']
    @server_id = ENV['MAILOSAUR_SERVER_ID']

    options = Selenium::WebDriver::Options.chrome(args: ['--headless=new'])
    @browser = Selenium::WebDriver.for :chrome, options: options

    # Instantiate Mailosaur client with api key
    @mailosaur = Mailosaur::MailosaurClient.new(api_key)
  end

  it 'Retrieves a one time passcode' do
    # Random test email address (this uses a catch-all pattern)
    random_string = ('a'..'z').to_a.shuffle[0,8].join
    email_address = "#{random_string}@#{@server_id}.mailosaur.net";

    # 1 - Request password reset
    @browser.get('https://example.mailosaur.com/otp')
    @browser.find_element(css: 'input#email').send_keys(email_address)
    @browser.find_element(css: 'button[type="submit"]').click

    # 2 - Create the search criteria for the email
    criteria = Mailosaur::Models::SearchCriteria.new()
    criteria.sent_to = email_address

    # 3 - Get the email from Mailosaur using the search criteria
    # https://mailosaur.com/docs/languages/ruby#2-messagesgetserver-criteria-options
    email = @mailosaur.messages.get(@server_id, criteria)

    # 4 - Extract the passcode from the email
    # https://mailosaur.com/docs/test-cases/codes
    passcode = email.html.codes[0]

    puts ("\nEmail otp code - #{passcode.value}")

    @browser.quit()
  end
end